GitHub: Update help.yml

joevaugh4n opened the pull request:
Closes #

<!-- If your PR is related to an issue, provide the number(s) above; if it resolves multiple issues, be sure to break them up (e.g. "closes #1000, closes #1001"). -->

<!--

Thank you for contributing to Storybook! Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `main` branch as part of the release process, so you shouldn't need to worry about this. For additional guidance: https://storybook.js.org/docs/react/contribute/how-to-contribute

-->


## What I did

- Fixed the help template not working
- Made tweaks to the layout (after seeing what it looked like in prod, wanted to change a few things around)

## Checklist for Contributors

### Testing

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to communicate how to test your changes -->

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

• Open up a help discussion
• Form should now work as syntax errors are resolved

### Documentation

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to indicate which documentation has been updated. -->

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

<!-- CANARY_RELEASE_SECTION -->

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

<!-- CANARY_RELEASE_SECTION -->

joevaugh4n committed:
Update help.yml

JReinhold commented on .github/DISCUSSION_TEMPLATE/help.yml:
```diff
@@ -1,25 +1,36 @@
-  - type: textarea
+  - type: markdown
+    attributes:
+      value: |
```
This text uses three different words for the same thing: discussion, issue and thread.

Can we use one of them?

joevaugh4n replied:
thanks, fair point @JReinhold - resolved!

joevaugh4n committed:
Update help.yml

kylegach commented on .github/DISCUSSION_TEMPLATE/help.yml:
```diff
@@ -1,25 +1,36 @@
+      description: |
+        Share Your Storybook configuration (`main.js or main.ts`), your Storybook version number, any error messages, or any relevant dependencies. These help us get a clearer understanding of what might be going wrong.
+
+        P.s. please share code as text rather than as a screenshot! It makes debugging much easier and faster.
```
```suggestion
        P.s. please [share code as text](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) rather than as a screenshot! It makes debugging much easier and faster.
```

kylegach commented on .github/DISCUSSION_TEMPLATE/help.yml:
```diff
@@ -1,25 +1,36 @@
       label: Additional information
-      description: Share your Storybook configuration (`main.js or ts`), any error messages, or relevant dependencies
+      description: |
+        Share Your Storybook configuration (`main.js or main.ts`), your Storybook version number, any error messages, or any relevant dependencies. These help us get a clearer understanding of what might be going wrong.
```
```suggestion
        Share Your Storybook configuration (`main.js` or `main.ts`), your Storybook version number, any error messages, or any relevant dependencies. These help us get a clearer understanding of what might be going wrong.
```

kylegach commented on .github/DISCUSSION_TEMPLATE/help.yml:
```diff
@@ -1,25 +1,36 @@
-  - type: textarea
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to start a new discussion!
+        
+        Before you post:
+        • Check if someone has already asked/answered your question in a previous discussion.
+        
+        When you're ready to post:
+        • Add tags to your discussion (e.g. 'React', 'Vue', 'Vite') to make it clearer for other users.
+        
```
```suggestion
      value: |
Thanks for taking the time to start a new discussion!
        
Before you post:
- Check if someone has already asked/answered your question in a previous discussion.
        
When you're ready to post:
- Add tags to your discussion (e.g. 'React', 'Vue', 'Vite') to make it clearer for other users.
        
```

Why?

This value (as well as `description`s, in other properties) is formatted as markdown. As such, indents of four or more spaces will likely result in that content being wrapped in code block:

_This markdown:_

> ```md
> Text
> 
>     Indented text
>  
>  More text
>  ```

 
 _Produces:_

> Text
>  
>  ```
>  Indented text
>  ```
>  
>  More text

-----

_If you already confirmed that your current content displays as desired, then disregard this comment!_

kylegach reviewed (commented):
Left some suggestions and comments to address. Ping me when you're ready, and I'll merge for ya!

joevaugh4n committed:
Update .github/DISCUSSION_TEMPLATE/help.yml

Co-authored-by: Kyle Gach <kyle.gach@gmail.com>

joevaugh4n committed:
Update .github/DISCUSSION_TEMPLATE/help.yml

Co-authored-by: Kyle Gach <kyle.gach@gmail.com>

joevaugh4n committed:
Update .github/DISCUSSION_TEMPLATE/help.yml

Co-authored-by: Kyle Gach <kyle.gach@gmail.com>

joevaugh4n commented:
> Left some suggestions and comments to address. Ping me when you're ready, and I'll merge for ya!

Hey @kylegach, thanks for the great feedback. Committed it. Unless there's anything else that I missed, please feel free to merge!

vanessayuenn commented on .github/DISCUSSION_TEMPLATE/help.yml:
```diff
@@ -1,25 +1,36 @@
+- Check if someone has already asked/answered your question in a previous discussion.
+        
+When you're ready to post:
+- Add tags to your discussion (e.g. 'React', 'Vue', 'Vite') to make it clearer for other users.
```
```suggestion
- Add labels to your discussion (e.g. 'React', 'Vue', 'Vite') to make it clearer for other users.
```
A nit to align with GitHub terminology

vanessayuenn reviewed (approved):
The content looks good to me. But it looks like [there are still some errors](https://github.com/storybookjs/storybook/blob/f74414b52351498413d8976727120516c595bc5f/.github/DISCUSSION_TEMPLATE/ideas.yml?plain=1) with the formatting 😬

joevaugh4n committed:
Update .github/DISCUSSION_TEMPLATE/help.yml

Co-authored-by: Vanessa Yuen <6842965+vanessayuenn@users.noreply.github.com>

joevaugh4n committed:
Update help.yml

joevaugh4n committed:
Update help.yml

joevaugh4n committed:
Update help.yml

joevaugh4n committed:
Update help.yml

joevaugh4n committed:
Update help.yml

joevaugh4n committed:
Update help.yml

joevaugh4n committed:
Update help.yml

joevaugh4n committed:
Update help.yml

joevaugh4n committed:
Update help.yml

joevaugh4n committed:
Update help.yml

joevaugh4n committed:
Update help.yml

joevaugh4n commented on .github/DISCUSSION_TEMPLATE/help.yml:
```diff
@@ -1,25 +1,36 @@
-      label: Share an example
-      description: Help us debug your issue by creating a minimal reproduction. You can do this with StackBlitz by heading to [https://storybook.new](https://storybook.new)!
+      label: Create a reproduction
+      description: Help us debug by creating a minimal reproduction with [https://storybook.new](https://storybook.new)
```
Hey @vanessayuenn – I tried the edited code in a different forked version of Storybook and it looks like it works (pic below). 

I can't get rid of the error messages in this branch – they don't even make sense because they refer to bits of code that aren't there. E.g. it says 'body[0]: validations is not a permitted attribute', but there isn't a validations attribute within that first key.

Any thoughts? Could we try merging and see if that resolves it/changes the errors? 

<img width="1265" alt="CleanShot 2023-11-17 at 14 10 01@2x" src="https://github.com/storybookjs/storybook/assets/122215197/6b3afb4d-ba34-44ff-bd52-be6c8af2e2db">

jonniebigodes committed:
Merge branch 'next' into joevaugh4n-sb-help

joevaugh4n committed:
Merge branch 'next' into joevaugh4n-sb-help

kylegach committed:
Update help.yml

- Remove `render` attribute from Additional information textarea
    - That property will force input into a code block, but we're asking for information that would best fit in multiple code blocks
- Add `id` for debugging purposes
- Attempt to fix formatting issues

kylegach committed:
Merge branch 'next' into joevaugh4n-sb-help

kylegach commented:
@joevaugh4n — I pushed a commit with some minor updates.

Looking more closely at those errors, they match the file that's on the `next` branch. It appears to be a bug in the GitHub UI that those errors appear when viewing a different branch, like this one. 🤷‍♂️ 

As far as I can tell, the file in this PR fixes all of those errors. So, I think you're good to go! I'm going to merge it.
